cmd/criticize/main.go

```diff
@@ -169,16 +177,25 @@ func (l *linter) CheckPackage(pkgPath string) {
 
 	l.ctx.SetPackageInfo(&pkgInfo.Info, pkgInfo.Pkg)
 	for _, f := range pkgInfo.Files {
-		l.ctx.SetFileInfo(l.getFilename(f))
-		l.checkFile(f)
+		if l.checkGenerated || !isGenerated(f) {
+			l.ctx.SetFileInfo(l.getFilename(f))
+			l.checkFile(f)
+		}
+	}
+}
+
+func isGenerated(f *ast.File) bool {
+	if f.Comments == nil || len(f.Comments) == 0 {
```

Comment:
Isn't nil check redundant here?

Reply:
Indeed. Hm...sounds like an idea for check. Will fill an issue ;D

Reply:
gosimple should catch this. I wonder why it doesn't trigger on this one.
Almost feeling uncomfortable because of this.